### PR TITLE
feat(annotations): support for suggest endpoint

### PR DESCRIPTION
**Pull request by peetcremer1**
- Added support for the new `/suggest` endpoint on the Annotations API
- Added a new interface `AnnotationSuggest` for it, which does not contain the `status` field as it is implicitly set to "suggested"
- Have the `AnnotationCreate` interface derive from `AnnotationSuggest``

### CODE REVIEW

File: packages/playground/src/api/annotations/types.ts [hunk 1]
```diff
@@ -24,7 +24,11 @@ export interface AnnotationModel extends AnnotationCreate {
   lastUpdatedTime: Date;
 }
 
-export interface AnnotationCreate {
```
Comment (miladcognite) [inline packages/playground/src/api/annotations/types.ts:27]: Outside the scope of this PR, but while you're at it :innocent: could you also fix `annotationType` in `AnnotationUpdate`.
Should be `annotationType?: SetField<AnnotationType>;`

**Review by miladcognite — APPROVED** [review]
LGTM :+1:

**Review by cyriaquebrousse — APPROVED** [review]

File: packages/playground/src/api/annotations/types.ts (hunk 1 above)
Comment (peetcremer1) [reply packages/playground/src/api/annotations/types.ts:27]: `AnnotationUpdate` is already like that :-) 
https://github.com/cognitedata/cognite-sdk-js/blob/master/packages/playground/src/api/annotations/types.ts#L46

However, there was another thing that was missing: `creatingAppVersion` in the `AnnotationFilterProps` which I added here: https://github.com/cognitedata/cognite-sdk-js/pull/806/commits/637e8c90c3ff8f70f919e14a25b428dab800d437

Comment (miladcognite) [reply packages/playground/src/api/annotations/types.ts:27]: hmm...shouldn't `annotationType?: SetField<AnnotationPayload>; ` -> `annotationType?: SetField<AnnotationType>; ` or am I missing something?

Comment (peetcremer1) [reply packages/playground/src/api/annotations/types.ts:27]: Ah yes, good catch!
Fixed it here: https://github.com/cognitedata/cognite-sdk-js/pull/806/commits/f3c9c3e5d9e00344450a430c02680a665268ae91

**Comment by peetcremer1** [comment]
Also made `annotatedResourceId` required on `AnnotationSuggest` and `AnnotationCreate` and deprecated `annotatedResourceExternalId`, as it is required by the API now in this commit: https://github.com/cognitedata/cognite-sdk-js/pull/806/commits/fd0c34243c186a019ec6c63d78f4ea9d60760965

**Review by marvrez — APPROVED** [review]

**Review by cognite-ornellas — APPROVED** [review]

**Pull request merged**